### PR TITLE
Removes set default, updates tests and readme

### LINKED ISSUE #13: Move "default widget src" replacement to bos-workspace
Since the development and merge of #4, and the progress with [bos-workspace](https://github.com/nearbuilders/bos-workspace) and it's relationship with this web component, the ability to set a default widget src no longer seems appropriate for this repository.

## Why?

1) The default route comes from `bos.config.json`, which is the configuration file for bos-workspace. Workspaces use near-bos-webcomponents to take advantage of the custom elements available through their custom VM's. I don't think `bos.config.json` should exist outside the context of a bos-workspace.
2) This replacement is currently occurring during webpack build, and isn't very convenient. It frequently interferes or overrides with development when using a custom web component in bos-workspace.
3) This replacement should be flexible according to network, and whether it is a dev build or a production deploy. bos-workspace has alias support, as well as separate dev and deploy commands.

## How?

Just as bos-workspace is currently using a `-g` flag to use a custom dist and [replaces rpc value with local proxy](https://github.com/NEARBuilders/bos-workspace/blob/92a66666ccd841223b84df62a287008f9547c93d/lib/gateway.ts#L30), we can use it to handle the replacement for the index widget and it's props, etc. 

## Conclusion

This should lead to a cleaner separation of duties and isolate this repository specific to the web component and it's functionality, and it's relationship to bos-workspace and other dev tools.

### PR DESCRIPTION
**Pull request by elliotBraem**
Resolves #13

**TLDR:** Removed set default widget from webpack build step here, and moved this ability to bos-workspace.

The ability to configure a default widget from `bos.config.json` has been added to bos-workspace via this [merged PR](https://github.com/NEARBuilders/bos-workspace/pull/96), via the "index" option in bos.config.json, [see docs here](https://github.com/NEARBuilders/bos-workspace?tab=readme-ov-file#base-configuration).

More work will be done to support index according to network and the deploy command. Otherwise, outside bos-workspace, devs can manually set the attributes in their own projects and not have webpack interfere.

---

# Context

Since the development and merge of #4, and the progress with [bos-workspace](https://github.com/nearbuilders/bos-workspace) and it's relationship with this web component, the ability to set a default widget src no longer seems appropriate for this repository.

## Why?

1) The default route comes from `bos.config.json`, which is the configuration file for bos-workspace. I don't think `bos.config.json` should exist outside the context of a bos-workspace, and I think this repository should be focused on the web component itself.
2) This replacement is currently occurring during webpack build, and isn't very convenient. It frequently interferes or overrides with development when using a custom web component in bos-workspace.
3) This replacement should be flexible according to network, and whether it is a dev build or a production deploy. bos-workspace has alias support, as well as separate dev and deploy commands.

## How?

Just as bos-workspace is currently using a `-g` flag to use a custom dist and [replaces rpc value with local proxy](https://github.com/NEARBuilders/bos-workspace/blob/92a66666ccd841223b84df62a287008f9547c93d/lib/gateway.ts#L30), we can use it to handle the replacement for the index widget and it's props, etc. 

(This change has been completed and merged, [see PR](https://github.com/NEARBuilders/bos-workspace/pull/96))

## Conclusion

This should lead to a cleaner separation of duties and isolate this repository specific to the web component and it's functionality, and it's relationship to bos-workspace and other dev tools.

### CODE REVIEW

**Review by petersalomonsen — APPROVED** [review]
Makes sense now that we can configure the same through the attributes of the component.
